http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -57,12 +55,15 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
     for {
       ctx <- as.ask
 
+      // Resolve the user email in Sam from the user token. This translates a pet token to the owner email.
+      userEmail <- samService.getUserEmail(userInfo.accessToken.token)
```

Comment:
This is something that Leo does wrong in a lot of places: we shouldn't use `userInfo.userEmail` because it won't translate the email address for pets. Instead, we should use `userInfo.accessToken`, and resolve the email in Sam when we need it.

I fixed it opportunistically in the create/deleteResource methods I touched, but there are still other places which use the old pattern. I can try to address the rest in subsequent PRs.

Reply:
Sounds good, I can note it in the acceptance criteria of https://broadworkbench.atlassian.net/browse/IA-5063. 

(That ticket might end up being multiple subtasks/PRs)